openeogeotrellis/job_tracker_v2.py

```diff
@@ -68,6 +80,72 @@ def get_job_metadata(self, job_id: str, user_id: str, app_id: str) -> _JobMetada
         raise NotImplementedError
 
 
+class JobCostsCalculator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def calculate_costs(self, job_info: dict, job_metadata: _JobMetadata, result_metadata: dict) -> float:
+        raise NotImplementedError
+
+
+class EtlApiJobCostsCalculator(JobCostsCalculator):
```

Comment:
If I understand correctly this in not only about calculating the cost, but also reporting/logging, right?

Reply:
Another thing: as far as I understand, the job cost calculation is quite separate from the job tracker itself. Could it make sense to move it  to its own module to have clearer separation of concerns?

Reply:
> If I understand correctly this in not only about calculating the cost, but also reporting/logging, right?

Yes: you give the ETL API some resource usage and it returns the costs in credits; then you patch those costs in the batch job entity.

> Another thing: as far as I understand, the job cost calculation is quite separate from the job tracker itself. Could it make sense to move it to its own module to have clearer separation of concerns?

Possibly. There's also the fact that a custom `JobCostsCalculator` implementation relies on the private `_JobMetadata` class so there's definitely room for improvement.

Reply:
I moved it to the `job_costs_calculator` module.